crates/ruff_linter/src/rules/flake8_type_checking/rules/runtime_import_in_type_checking_block.rs

```diff
@@ -154,7 +154,9 @@ pub(crate) fn runtime_import_in_type_checking_block(
                 if checker.settings.flake8_type_checking.quote_annotations
                     && binding.references().all(|reference_id| {
                         let reference = checker.semantic().reference(reference_id);
-                        reference.in_typing_context() || reference.in_runtime_evaluated_annotation()
+                        reference.in_typing_context()
+                            || reference.in_runtime_evaluated_annotation()
+                            || reference.in_explicit_type_alias()
                     })
```

Comment:
I suspect this change behavior for existing rules. If so, then I think it would make sense to extract this change into its own PR so that it gets highlighted in the changelog.

Reply:
Fair point. Although you'd get infinite fix/violation cycles without this extra check between the TCH001-004 and TCH007-008 rules. So maybe I should only perform this extra check if TCH007/TCH008 are enabled. That way the behavior stays the same unless you enable the preview rules.

Reply:
Actually this doesn't seem to cause unstable fixes after all, I think I changed this so the behavior is more consistent with the flake8 plugin.

I'll remove this change for now and leave a comment. I think eventually we want a setting to independently configure the preferred action for runtime annotations and explicit type aliases, since depending on the use-case you want the type alias to be available at runtime, but you may not care about runtime annotations.